### ISSUE TITLE
Slow training performance on new multi-gpu system

### ISSUE THREAD

**Issue by cgpadwick** [weiliu89/caffe#691]
I have an issue I wanted to share with group.  I have been using CaffeSSD successfully for over 1 year now.  I have two platforms in play:

Original Platform: Ubuntu 14.04, NVIDIA Driver: 370.28, 4X Titan X (Pascal)
New Platform: Ubuntu 16.04, NVIDIA Driver: 378.13, 8X 1080 Ti (Pascal)

On the original platform, multi-gpu training works fine.  On the new platform, the training goes very slowly.  To test, I did a 4 GPU training run on each platform.  Here is a log excerpt:

New Platform
I0727 18:04:08.804803 39460 solver.cpp:243] Iteration 4910, loss = 4.56288
I0727 18:04:38.560071 39460 solver.cpp:243] Iteration 4920, loss = 3.90785
delta T = 30 seconds

Original Platform
I0715 12:32:08.124413 12547 sgd_solver.cpp:106] Iteration 39920, lr = 1e-05
I0715 12:32:09.693231 12547 solver.cpp:231] Iteration 39930, loss = 7.67515
delta T = 1.5 seconds

The training takes an extremely long time on the new platform.  The GPU utilization on the new platform is very low during training, much lower than the original platform.  in both tests the lmdb is being served from local solid state storage.  I have tested the new platform with a tensorflow benchmark and it works very well, and the GPU utilization is at 100% for all 8 GPU's for the duration of the test.  Evidence points to some problem with the new driver with 1080 Ti and this version of Caffe.

Has anyone seen this behavior before?  We can't roll back to the 370.28 driver because it doesn't support the 1080 Ti.  Tried rolling forward as well to the newest driver, but hat didn't make any difference.  

I notice that caffe master branch now uses nccl.  nccl is specifically designed to improve map/reduce/broadcast operations over the PCI bus.  I'm wondering if the new driver for the 1080 Ti is causing problems with non-nccl code.  I thought I might try merging the ssd branch into BVLC caffe master and seeing if using nccl support fixes the issue.

I see that there have been a couple of  attempts:

https://github.com/BVLC/caffe/pull/4965 which was closed with no explanation
https://github.com/BVLC/caffe/pull/5806 which is appears to be brand new, but has some conflicts

I'm having trouble understanding why ssd hasn't been brought into the master branch of ssd, at one point it was nearly a trivial merge.  Anyway, any thoughts would be greatly appreciated.
